pandas/formats/format.py

```diff
@@ -2138,6 +2016,31 @@ def formatter(value):
 
         return formatter
 
+    def _get_formatted_data(self):
+        """
+        Returns formatted float values. This is different from
```

Comment:
I think my reasoning was made clear in the doc-string.  `get_result_as_array` returns only strings (it supposed to be doing this for quite some time).  However, formatting CSV output is **not always** string.